### PR TITLE
Expand box_vec lint to box_collection

### LINKED ISSUE #7451: Suggest box_string
### What it does

Check for usage for `Box<String>` in struct definition and function parameter.

### Categories (optional)

Perf, Warn

*What is the advantage of the recommended code over the original code*

`String` already allocates on heap, `Box<String>` allocates a pointer on heap.
Similar lints:  `rc_buffer`, `box_vec`, `redundant_allocation`,

### Drawbacks

None.

### Example

```rust
struct foo(Box<String>)
fn bar(a: Box<String>){}
```

Could be written as:

```rust
struct foo(String)
fn bar(a: String){}
```

### PR DESCRIPTION
**Pull request by F3real**
fixed #7451 

changelog: Expand `box_vec` into [`box_collection`], and have it error on all sorts of boxed collections

### CODE REVIEW

**Comment by F3real** [comment]
@flip1995 it seems that `remark` failure is related to mention of `box_vec` in CHANGELOG.md. Should we also remove mentions from changelog? It feels a bit weird.

File: clippy_lints/src/lib.rs [hunk 1]
```diff
@@ -956,7 +956,7 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
         transmuting_null::TRANSMUTING_NULL,
         try_err::TRY_ERR,
         types::BORROWED_BOX,
-        types::BOX_VEC,
+        types::BOX_COLLECTION,
```
Comment (Manishearth) [inline clippy_lints/src/lib.rs:959]: also should add this rename to `register_renamed` below

**Review by Manishearth — CHANGES_REQUESTED** [review]

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
@F3real you don't have to remove mentions, you just need to make sure the `box_vec` link reference exists and links to `box_collections`

**Review by Manishearth — APPROVED** [review]

**Pull request merged**